daemon/monitor.go

```diff
@@ -90,7 +93,7 @@ func (daemon *Daemon) StateChanged(id string, e libcontainerd.StateInfo) error {
 		daemon.setStateCounter(c)
 
 		defer c.Unlock()
```

Comment:
Unrelated to these changes, but it's odd that this line isn't up near the call to `Lock`.